pkgs/by-name/cu/cusf-predictor-wrapper/package.nix

```
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  glib,
+  installShellFiles,
+}:
+stdenv.mkDerivation {
+  pname = "cusf_predictor_wrapper";
+  version = "0-unstable-2025-03-04";
+
+  src = fetchFromGitHub {
+    owner = "darksidelemm";
+    repo = "cusf_predictor_wrapper";
+    rev = "f4352834a037e3e2bf01a3fd7d5a25aa482e27c6";
+    hash = "sha256-C8/5x8tim6s0hWgCC7LpN1hesdVME5kpQFqDTEyXHtg=";
+  };
+
+  sourceRoot = "source/src";
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    glib
+    installShellFiles
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mv pred cusf_predictor_wrapper
+    installBin cusf_predictor_wrapper
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "CUSF standalone predictor";
+    homepage = "https://github.com/darksidelemm/cusf_predictor_wrapper";
+    license = lib.licenses.gpl3;
```

```suggestion
    license = lib.licenses.gpl3Only;
```
deprecated

Also, [some files](https://github.com/darksidelemm/cusf_predictor_wrapper/blob/f4352834a037e3e2bf01a3fd7d5a25aa482e27c6/src/util/getline.c#L4-L7) are licensed under the gpl2Plus and [others](https://github.com/darksidelemm/cusf_predictor_wrapper/tree/f4352834a037e3e2bf01a3fd7d5a25aa482e27c6/src/ini) are licensed under the mit.

---

How do we want to resolve the different files having different licenses? Is it safe to use gpl3only since that's what the overarching repo is marked as?